python/pyarrow/array.pxi

```diff
@@ -988,12 +988,64 @@ cdef class Array(_PandasConvertible):
         """
         Total number of bytes consumed by the elements of the array.
         """
+        return self.get_refererenced_buffer_size()
+
+    def get_total_size_by_buffer(self):
```

Comment:
How is this now different from `get_total_buffer_size`?

Reply:
Good catch. 

This is a question I also have. Here the Python method counts the elements as same as it is done in the C++ method `TotalBufferSize`. For now I kept it as this is still WIP. I also felt like we must remove the calculation done in Python and just use the C++ method. Am I following this right?

Reply:
I added a discussion in the ticket while going through the code

https://issues.apache.org/jira/browse/ARROW-15065

Reply:
I have added both `TotalBufferSize` and `ReferencedBufferSize` with test cases.